### PR TITLE
Modularise discovery

### PR DESCRIPTION
**Pull request by awiouy**

### CODE REVIEW

File: discovery/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+base64 = "0.5.0"
+futures = "0.1.8"
+hyper = "0.11.2"
+log = "0.3.5"
+num-bigint = "0.1.35"
+protobuf = "1.1"
+rand = "0.3.13"
+rust-crypto = { git = "https://github.com/awmath/rust-crypto.git", branch = "avx2" }
+serde = "0.9.6"
+serde_derive = "0.9.6"
```
Comment (sashahilton00) [inline discovery/Cargo.toml]: Why are ```serde``` and ```serde_derive``` required in this crate?

**Comment by sashahilton00** [comment]
LGTM, other than the redundant dependencies.

File: discovery/Cargo.toml (hunk 1 above)
Comment (awiouy) [reply discovery/Cargo.toml]: https://github.com/awiouy/librespot/blob/f91a380ea43b11d3caa4622e149c23b8a6f5cc4f/discovery/src/discovery.rs#L29

**Comment by awiouy** [comment]
Which redundant dependencies?

File: discovery/Cargo.toml (hunk 1 above)
Comment (sashahilton00) [reply discovery/Cargo.toml]: ok, serde required, missed that, but what about serde_derive? I don't see it in use anywhere

Comment (awiouy) [reply discovery/Cargo.toml]: `#[derive(Clone)]`
https://serde.rs/derive.html
We might (try to) remove serde-json later on

Comment (sashahilton00) [reply discovery/Cargo.toml]: ok, fair enough. Not sure why they've kept that seperate...

**Comment by sashahilton00** [comment]
None apparently :) Merging.

**Comment by plietar** [comment]
Is there a reason this PR was done against a separate branch and not master?
Give the high risk of conflicts (because of the file rename) this should go into master ASAP.

**Comment by sashahilton00** [comment]
Originally I forked the master branch t owork on modularising things, and by coincidence Anton and I started working at modularising librespot at the same time. I was going to modularise the audio_backend and spirc as well before merging, but given the potential changes in https://github.com/librespot-org/librespot/pull/125, it's probably sensible to merge the progress thus far sooner rather than later. Will let the CI run, then merge this to master.

**Pull request merged**